### DISCUSSION TITLE
Upgraded to 3.10.0.CR1 failed In-Memory Message Tests

### DISCUSSION THREAD

**Discussion by hantsy** [quarkusio/quarkus#40219] [Q&A]
All messaging examples in my repository are failed when upgrading to the newest Quarkus 3.10.0.CR1, eg. https://github.com/hantsy/quarkus-sandbox/tree/master/amqp

I followed this guide to clean up the dependency artifact id and use the new name in all messaging examples, https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.9#resteasy-reactive-extensions-renamed-to-quarkus-rest-gear-white_check_mark

```java
@QuarkusTest
@TestProfile(InMemoryProfile.class)
class MessageHandlerTest {

    @Inject
    @Any
    InMemoryConnector connector;

    @Inject
    MessageHandler handler;

    @BeforeEach
    void setUp() {
    }

    @Test
    void receive() {
        InMemorySource<String> messages = connector.source("messages");
        InMemorySink<String> sink = connector.sink("send");
        InMemorySink<Message> dataStream = connector.sink("data-stream");


        handler.send("hello");
        assertThat(sink.received().getFirst().getPayload()).isEqualTo("hello");

        messages.send("hello-123");
        assertThat(dataStream.received().getFirst().getPayload().body()).isEqualTo("hello-123");
    }
}

```

The `assertThat(dataStream.received().getFirst().getPayload().body()).isEqualTo("hello-123");` throws the following exceptions

```bash
java.util.NoSuchElementException
	at java.base/java.util.ArrayList.getFirst(ArrayList.java:439)
	at com.example.MessageHandlerTest.receive(MessageHandlerTest.java:41)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:1018)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestMethod(QuarkusTestExtension.java:832)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)

2024-04-23 22:38:03,206 INFO  [io.quarkus] (main) quarkus-example-rabbitmq stopped in 0.033s

Process finished with exit code -1
```

**Comment by hantsy** [comment]
In the kafka and pulsar example, the same tests got different errors: https://github.com/hantsy/quarkus-sandbox/actions/runs/8800974801/job/24153288400

```bash 
Invalid channel configuration -  the `connector` attribute must be set for channel `data-stream`
```

**Reply by ozangunalp** [reply]
I don't think it ever worked. In that test `data-stream` is an internal channel. 
You can only replace channels wired to a connector with the in-memory connector.

**Reply by hantsy** [reply]
Check the 3.7.2 build, https://github.com/hantsy/quarkus-sandbox/actions/runs/7881130800/job/21504268906, amqp, amqp-rabbit, rabbit worked well.

But pulsar in-memory example did not work due to an issue I have not found a solution. see: https://github.com/quarkusio/quarkus/discussions/36764

**Comment by gsmet** [comment]
I created https://github.com/quarkusio/quarkus/issues/40317 as it looks odd.
